metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/samples/DefaultInstanceSampleSpecModsProvider.java

```diff
@@ -24,22 +26,81 @@ public DefaultInstanceSampleSpecModsProvider addMod(
         MetricNamedPredicateBuilder<?> predicateBuilder,
         InstanceSampleSpecProvider<DefaultInstanceSampleSpec> mod) {
 
-        return addMod(predicateBuilder.build(), mod);
+        return addMod(null, predicateBuilder, mod);
+    }
+
+    /**
+     * Adds the specified modification by the specified key.
+     * You can further remove the modification using the {@link #removeMod} method after it's no longer needed.
+     *
+     * @param key key with which the modification is to be associated.
+     *            May be null, in which case there will be no association, and you won't be able to remove the modification.
+     */
+    public DefaultInstanceSampleSpecModsProvider addMod(
+        String key,
+        MetricNamedPredicateBuilder<?> predicateBuilder,
+        InstanceSampleSpecProvider<DefaultInstanceSampleSpec> mod) {
+
+        return addMod(key, predicateBuilder.build(), mod);
     }
 
     public DefaultInstanceSampleSpecModsProvider addMod(
         MetricNamedPredicate predicate,
         InstanceSampleSpecProvider<DefaultInstanceSampleSpec> mod) {
 
-        return addInfo(predicate, mod);
+        return addMod(null, predicate, mod);
+    }
+
+    /**
+     * Adds the specified modification by the specified key.
+     * You can further remove the modification using the {@link #removeMod} method after it's no longer needed.
+     *
+     * @param key key with which the modification is to be associated.
+     *            May be null, in which case there will be no association, and you won't be able to remove the modification.
+     */
+    public DefaultInstanceSampleSpecModsProvider addMod(
+        String key,
+        MetricNamedPredicate predicate,
+        InstanceSampleSpecProvider<DefaultInstanceSampleSpec> mod) {
+
+        return addInfo(key, predicate, mod);
     }
 
     @Override
     public DefaultInstanceSampleSpecModsProvider addInfo(
         MetricNamedPredicate predicate,
-        InstanceSampleSpecProvider<DefaultInstanceSampleSpec> mod) {
+        InstanceSampleSpecProvider<DefaultInstanceSampleSpec> info) {
 
-        parent.addInfo(predicate, mod);
+        return addInfo(null, predicate, info);
+    }
+
+    /**
+     * Adds the specified modification by the specified key.
```

Comment:
Could we use here?
```
 /**
    * {@inheritDoc}
    */
```
It allows to avoid duplication and, IMHO, improves comprehensibility (read once the parent class, understand immediately that it's identical to it. Otherwise, you should read through this comment to find the difference between two).

Reply:
Agreed, thank you!

Reply:
I'll also consider adding the corresponding methods in those interfaces.